pkg/apis/core/validation/validation.go

```diff
@@ -4688,6 +4689,17 @@ func ValidatePodEphemeralContainersUpdate(newPod, oldPod *core.Pod, opts PodVali
 	allErrs = append(allErrs, validatePodMetadataAndSpec(newPod, opts)...)
 	allErrs = append(allErrs, ValidatePodSpecificAnnotationUpdates(newPod, oldPod, fldPath.Child("annotations"), opts)...)
 
+	// static pods don't support ephemeral containers #113935
+	v1PodOld := &v1.Pod{}
+	// TODO check for error below
+	// how to handle this error?!
+	// Its not appropriate to send it on to the user
+	// they can't fix it
+	corev1.Convert_core_Pod_To_v1_Pod(oldPod, v1PodOld, nil)
```

Comment:
why do you need to convert? to use `types.IsMirrorPod`? no need, check  `ValidatePodSpecificAnnotationUpdates` , it already checks for mirror pod, just copy the logic

Reply:
I remember why. I thought it best to use the `types.IsMirrorPod` since I thought it best to avoid c/p.  Its fixed now.  Thanks!